### PR TITLE
K8SPS-202 fix the restoration of the backup

### PR DESCRIPTION
**Pull request by hors**
[K8SPS-202](https://jira.percona.com/browse/K8SPS-202) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

* We have cases when replica source should be changed during the restore process e.g. primary was pod-2 and replicas were pod-0 and pod-1. So, during the restore typology will be changed to pod-0 (primary) and pod-1 and pod-2 will be
secondary. That is why during the bootstrap we need to change 'CHANGE REPLICATION SOURCE' but this operation cannot be performed with running replication threads. We need to stop it before we change the replication source.

### CODE REVIEW

File: pkg/replicator/replicator.go [hunk 1]
```diff
@@ -98,6 +98,10 @@ func (d *dbImpl) ChangeReplicationSource(host, replicaPass string, port int32) e
 }
 
 func (d *dbImpl) StartReplication(host, replicaPass string, port int32) error {
+	if err := d.StopReplication(); err != nil {
```
Comment (hors) [inline pkg/replicator/replicator.go:101]: Maybe I need to move it to bootstrap/main.go ... What do you think?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/auto-config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/sidecars.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/gr-demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/one-pod.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/0f2aaac/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/225/commits/0f2aaac61e6655935bed38e953ce404e1570d763
image: `perconalab/percona-server-mysql-operator:PR-225-0f2aaac`

File: pkg/replicator/replicator.go (hunk 1 above)
Comment (egegunes) [reply pkg/replicator/replicator.go:101]: it may be a good idea but please provide a description, why does this fix the issue?

Comment (hors) [reply pkg/replicator/replicator.go]: @egegunes I have updated description

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/init-deploy.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/auto-config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/one-pod.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/tls-cert-manager.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/gr-demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/84837af/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/225/commits/84837af19ffc240a99aaf6920660eb9c509bafd4
image: `perconalab/percona-server-mysql-operator:PR-225-84837af`

**Review by inelpandzic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Review by pooknull — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/sidecars.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/gr-demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/limits.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/one-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/scaling.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-225/02611f5/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/225/commits/02611f54c232905804f219fb85cc4be77d6730e4
image: `perconalab/percona-server-mysql-operator:PR-225-02611f5`

**Pull request merged**